draft-thomson-http-oblivious.md

```diff
@@ -264,15 +306,131 @@ Message encrypted message.
+including identifying information. Similarly, the request that is sent to the
+oblivious request resource, though this request can contain only minimal
+information as it only needs to include a method and the oblivious request
+resource URL.
+
+
+## Proxy Responsibilities
+
+The proxy that serves the oblivious proxy resource has a very simple function
+to perform. It forwards messages received at this resource to the oblivious
+request resource.
+
+The proxy MUST NOT add information about the client identity when forwarding
+requests. This includes the Via field, the Forwarded field
+{{?FORWARDED=RFC7239}}, and any similar information.
+
+
+### Denial of Service {#dos}
+
+As there are privacy benefits from having a large rate of requests forwarded by
+the same proxy (see {{ta}}), servers that operate the oblivious request
+resource might need an arrangement with proxies. This arrangement might be
+necessary to prevent having the large volume of requests being classified as an
+attack by the server.
+
+If a server does accept a large volume of requests from a proxy, it needs to
+trust that the proxy does not allow abusive levels of request volumes from
+clients. That is, if a server allows requests from the proxy to be exempt from
+rate limits, the server might want to ensure that the proxy applies similar
+rate limiting when receiving requests from clients.
+
+Servers that enter into an agreement with a proxy that enables a higher request
+rate might choose to authenticate the proxy to enable the higher rate.
+
+
+### Linkability Through Traffic Analysis {#ta}
+
+As the time at which encapsulated request or response messages are sent can
+reveal information to a network observer. Though messages exchanged between the
+oblivious proxy resource and the oblivious request resource might be sent in a
+single connection, traffic analysis could be used to match messages that are
+forwarded by the proxy.
+
+A proxy could, as part of its function, add delays in order to increase the
+anonymity set into which each message is attributed. This could latency to the
+overall time clients take to receive a response, which might not what some
+clients want.
+
+A proxy can use padding to reduce the effectiveness of traffic analysis.
+
+A proxy that forwards large volumes of exchanges can provide better privacy by
+providing larger sets of messages that need to be matched.
+
+What steps a proxy takes to mitigate the risk of traffic analysis is something
+a client needs to consider when selecting a proxy.
+
+
+## Server Responsibilities
+
+A server that operates both oblivious request and oblivious target resources is
+responsible for removing request encapsulation, generating a response the
+encapsulated request, and encapsulating the response.
 
-## Oblivious Request Resource
+A server needs to ensure that the encapsulated responses it sends are padded to
+protect against traffic analysis; see {{ta}}.
```

Comment:
Is this enough? What if the response is leaked not via size but via response time? (Victim makes a request whose response is cached, and then attacker sends the same request and tries to observe cache hit/miss.) 

(I really don't love the timing delay stuff, but it seems better to mention the possible risk than not.)

```suggestion
Servers should account for traffic analysis based on response size or generation time.
Techniques such as padding or timing delays can help protect against such attacks;
see {{ta}}.
```